http_requests.go

```diff
@@ -12,35 +13,57 @@ import (
 	"github.com/gorilla/websocket"
 )
 
-func connectToWebsocket(wsURL string, reconnectToken uuid.UUID, accessToken string, subscriptionIDOrName string) (*websocket.Conn, error) {
-	URL := wsURL + "?"
-	URL = URL + "access_token=" + accessToken
-	URL = URL + "&subscription_id=" + subscriptionIDOrName
+type WebsocketSetupHTTPError struct {
+	error
+	HttpStatus int
+}
+
+var httpClient = &http.Client{
+	Timeout: time.Second * 10,
+}
+
+func connectToWebsocket(wsURL string, reconnectToken uuid.UUID, subscriptionIDOrName string) (*websocket.Conn, error) {
+	URL := wsURL + "?subscription_id=" + subscriptionIDOrName
 	if reconnectToken != uuid.Nil {
 		URL = URL + "&reconnect_token=" + reconnectToken.String()
 	}
+
+	// Add the auth credentials to the ws connection setup request
+	var h http.Header
+	if *clientV3SecretFlag != "" {
+		// Set the Abios secret as a header in the request
+		h = make(http.Header)
+		h["Abios-Secret"] = []string{*clientV3SecretFlag}
+	} else {
+		accessToken, err := requestAccessToken(*clientV2IDFlag, *clientV2SecretFlag)
+		if err != nil {
+			return nil, fmt.Errorf("Access token request failed. Error: %v", err)
+		}
+
+		URL = URL + "&access_token=" + accessToken
+	}
+
 	var dialer *websocket.Dialer
-	conn, resp, err := dialer.Dial(URL, nil)
-
-	if err == websocket.ErrBadHandshake {
-		fmt.Printf("%s [ERROR]: Failed to connect to WS url. Handshake status='%d'\n",
-			time.Now().Format(timestampMillisFormat), resp.StatusCode)
-		return nil, &WebsocketSetupHTTPError{HttpStatus: resp.StatusCode}
-	} else if err != nil {
-		fmt.Printf("%s [ERROR]: Failed to connect to WS url. Error='%s'\n",
-			time.Now().Format(timestampMillisFormat), err.Error())
-		return nil, err
+	conn, resp, err := dialer.Dial(URL, h)
+
+	if err != nil {
+		if resp != nil {
+			return nil, WebsocketSetupHTTPError{HttpStatus: resp.StatusCode, error: err}
+		} else {
```

Comment:
`else` can be removed.

Reply:
Let's keep it as is, but your comment made me realize I had forgotten to handle the error correctly in the caller. :+1: